app/src/androidTest/java/com/example/quizacademytask/MigrationTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.quizacademytask
+
+import androidx.room.migration.Migration
+import androidx.room.testing.MigrationTestHelper
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
+import androidx.test.platform.app.InstrumentationRegistry
+import db.AppDatabase
+import junit.framework.Assert.*
+import org.junit.Rule
+import org.junit.Test
+
+class MigrationTest {
+    private val testDBName = "test_db"
+
+    @JvmField
+    @Rule
+    val testHelper: MigrationTestHelper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        AppDatabase::
+        class.java.canonicalName, FrameworkSQLiteOpenHelperFactory()
+    )
+
+    private val MIGRATION_1_2: Migration = object : Migration(1, 2) {
+        override fun migrate(database: SupportSQLiteDatabase) {
+            database.execSQL("DROP TABLE IF EXISTS course;")
+            database.execSQL("ALTER TABLE `cardStack` RENAME TO `cardStackOLD`")
+            database.execSQL(
```

Comment:
basically you are deleting all card stack data here as it is not being copied from the old table to the new one. Migration were user looses their data is not very optimal.